plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/util/Util.java

```diff
@@ -60,4 +65,23 @@ public static String getFolder( String path ) {
     }
     return path.substring( 0, path.lastIndexOf( "/" ) );
   }
+
+  public static void rawElementMassage( Object rawElement ) {
+    //Performs conversion of raw elements from drag and drop to something the system understands
+    if ( rawElement instanceof VFSFile ) {
+      VFSFile vfsFile = (VFSFile) rawElement;
+      if ( !vfsFile.getPath().startsWith( "hcp" ) ) {
+        //HCP Files come back like this "hcp://hcp.hitachi.com:443/path1/filename" and for some reason this does not
+        // work with HCP.  We work around this by converting the URL to pvfs like this
+        // ( ie: pvfs://myconnectionName/path1/filename ).  The displayed name is what we want.
+        try {
+          URI uri = new URI( vfsFile.getPath() );
+          vfsFile.setPath( ConnectionFileProvider.SCHEME + "://" + vfsFile.getConnection() + uri.getPath() );
+          vfsFile.setParent( Util.getFolder( vfsFile.getPath() ) );
+        } catch ( URISyntaxException e ) {
+          //Nothing to do.  Uri should always work
```

Comment:
We should probably re-throw this and let the caller at least log it.  "Should always work" means "is going to fail when we least expect it and will be impossible to debug because we threw away the exception"